### PR TITLE
Prepare support for sqlx

### PR DESCRIPTION
**Pull request by agnivade**
Just setting up some basic skeleton to start
replacing gorp queries with sqlx.

There is still an issue with efficiently writing
cross-platform named queries, because :FieldName
needs to be :fieldname for Postgres.

The positional params can be rebound depending on
driver name. But named queries can't from my
investigation. Will look into this.

```release-note
NONE
```

### CODE REVIEW

**Comment by jespino** [comment]
Two small concerns that I'm having now are:

1. We are missing with this change the query timeout (I guess we are going to address that with something like QueryXContext, making that explicit to every query, or creating a `queryX` function in the store to address that (but is important to not forget about it).
2. We are missing also the tracing, that is not supported in sqlx, and I don't know how we are going to address that (there is a fork from DataDog that does exactly that, but I don't know how up-to-date it is).

Other than that, the PR looks good to me.

**Comment by agnivade** [comment]
Excellent points @jespino ! I will note them and appropriately come up with a solution. Thank you!

**Review by jespino — APPROVED** [review]
LGTM

File: store/sqlstore/store.go [hunk 1]
```diff
@@ -137,11 +138,18 @@ type SqlStoreStores struct {
...
+	srCounter int64
+
+	master  *gorp.DbMap
+	masterX *sqlx.DB
+
+	Replicas  []*gorp.DbMap
+	ReplicaXs []*sqlx.DB
+
+	searchReplicas  []*gorp.DbMap
+	searchReplicaXs []*sqlx.DB
```
Comment (streamer45) [inline store/sqlstore/store.go:151]: Is the plan to keep both (gorp and sqlx) for a while or just during development?

**Review by streamer45 — APPROVED** [review]

File: store/sqlstore/store.go (hunk 1 above)
Comment (agnivade) [reply store/sqlstore/store.go:151]: Yes, the plan is to keep them both during the transition period. And then remove them, and change all (Master|Replica)X methods to (Master|Replica) ones.

Comment (streamer45) [reply store/sqlstore/store.go:151]: Sounds good :+1:

**Comment by agnivade** [comment]
Gentle ping @isacikgoz / @mgdelacroix. I have the migration PRs waiting to be sent out. And there's the other [PR](https://github.com/mattermost/mattermost-server/pull/18148) in draft stage.

**Review by isacikgoz — APPROVED** [review]
LGTM 👍

**Review by mgdelacroix — APPROVED** [review]
LGTM!! :+1:

**Pull request merged**